workers/data_refinery_workers/processors/smasher.py

```diff
@@ -527,111 +627,28 @@ def _smash(job_context: Dict, how="inner") -> Dict:
-                        # So we use this, which also helps us support the case of missing SampleComputedFileAssociation
-                        data.columns = [sample.accession_code]
-                    except ValueError as e:
-                        # This sample might have multiple channels, or something else.
-                        # Don't mess with it.
-                        logger.exception("Smasher found multi-channel column (probably) - skipping!",
-                            computed_file_path=computed_file_path
-                        )
-                        unsmashable_files.append(computed_file.filename)
-                        continue
-                    except Exception as e:
-                        # Okay, somebody probably forgot to create a SampleComputedFileAssociation
-                        # Don't mess with it.
-                        logger.exception("Smasher found very bad column title - skipping!",
-                            computed_file_path=computed_file_path
-                        )
-                        unsmashable_files.append(computed_file.filename)
-                        continue
-
-                    if computed_file_path.endswith("lengthScaledTPM.tsv"):
-                        technologies_rnaseq[all_frames_index] = data.columns
-                    else:
-                        technologies_microarray[all_frames_index] = data.columns
-
-                    all_frames[all_frames_index] = data
-                    all_frames_index += 1
-                    num_samples = num_samples + 1
-
-                    if (num_samples % 100) == 0:
-                        logger.debug("Loaded " + str(num_samples) + " samples into frames.",
-                            dataset_id=job_context['dataset'].id,
-                            how=how
-                        )
-
-                except Exception as e:
-                    unsmashable_files.append(computed_file_path)
-                    logger.exception("Unable to smash file",
-                        file=computed_file_path,
-                        dataset_id=job_context['dataset'].id,
-                        )
-                finally:
-                    # Delete before archiving the work dir
-                    if computed_file_path:
-                        os.remove(computed_file_path)
-
-            all_frames = all_frames[0:all_frames_index]
+            start = datetime.now()
+            cpus = psutil.cpu_count()/2
+            with multiprocessing.Pool(int(cpus)) as pool:
+
+                processed_frames = pool.map(
+                    process_frame,
+                    [(
+                        job_context, 
+                        computed_file, 
+                        sample, 
+                        i
+                    ) for i, (computed_file, sample) in enumerate(input_files)
+                ])
+           
+            logger.debug(datetime.now() - start)
```

Comment:
Any reason you're doing it like this instead of using `log_state`?

Reply:
yeah ill be removing that, i have since changed it will push soon 👍